src/main/java/com/ldtteam/structurize/placement/handlers/placement/PlacementHandlers.java

```diff
@@ -183,27 +185,21 @@ public void handleRemoval(
 
         @Override
         public ActionProcessingResult handle(
-          Level world,
-          BlockPos pos,
-          BlockState blockState,
-          @Nullable CompoundTag tileEntityData,
-          boolean complete,
-          BlockPos centerPos)
+            Level world,
+            BlockPos pos,
+            BlockState blockState,
+            @Nullable CompoundTag tileEntityData,
+            boolean complete,
+            BlockPos centerPos,
+            RotationMirror settings)
         {
             if (complete)
             {
-                world.setBlock(pos, ModBlocks.blockFluidSubstitution.get().defaultBlockState(), UPDATE_FLAG);
+                handleBlockPlacement(world, pos, ModBlocks.blockFluidSubstitution.get().defaultBlockState());
                 return ActionProcessingResult.PASS;
             }
 
-            if (world.getBlockState(pos).hasProperty(BlockStateProperties.WATERLOGGED))
-            {
-                world.setBlock(pos, world.getBlockState(pos).setValue(BlockStateProperties.WATERLOGGED, true), UPDATE_FLAG);
```

Comment:
This is for the case of: If there is a block in the world that can be waterlogged, then waterlog if the block is fluid substitution you're supposed to place.